plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,276 @@
+		if err != nil {
+			return err
+		}
+		n.client = client
+	}
+
+	for _, city := range n.Cities {
+		var u *url.URL
+		var addr *url.URL
+
+		u, err = url.Parse(fmt.Sprintf("/data/2.5/forecast?id=%s&APPID=%s", city, n.AppId))
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+			continue
+		}
+		addr = base.ResolveReference(u)
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			acc.AddError(n.gatherUrl(addr, acc))
+		}(addr)
+
+		u, err = url.Parse(fmt.Sprintf("/data/2.5/weather?id=%s&APPID=%s", city, n.AppId))
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+			continue
+		}
+
+		addr = base.ResolveReference(u)
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			acc.AddError(n.gatherUrl(addr, acc))
+		}(addr)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *OpenWeatherMap) createHttpClient() (*http.Client, error) {
+
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{},
+		Timeout:   n.ResponseTimeout.Duration,
+	}
+
+	return client, nil
+}
+
+func (n *OpenWeatherMap) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	resp, err := n.client.Get(addr.String())
+
+	if err != nil {
+		return fmt.Errorf("error making HTTP request to %s: %s", addr.String(), err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s returned HTTP status %s", addr.String(), resp.Status)
+	}
+	contentType := strings.Split(resp.Header.Get("Content-Type"), ";")[0]
+	switch contentType {
+	case "application/json":
+		body, _ := ioutil.ReadAll(resp.Body)
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		err = gatherStatusUrl(bufio.NewReader(resp.Body), n.getTags(addr), acc)
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		err = gatherWeatherUrl(bufio.NewReader(resp.Body), n.getTags(addr), acc)
+		return err
+	default:
+		return fmt.Errorf("%s returned unexpected content type %s", addr.String(), contentType)
+	}
+}
+
+func (n *OpenWeatherMap) getTags(addr *url.URL) map[string]string {
+	h := addr.Host
+	host, port, err := net.SplitHostPort(h)
+	if err != nil {
+		host = addr.Host
+		if addr.Scheme == "http" {
+			port = "80"
+		} else if addr.Scheme == "https" {
+			port = "443"
+		} else {
+			port = ""
+		}
+	}
+	tags := map[string]string{
+		"server":   host,
+		"port":     port,
+		"base_url": n.BaseUrl,
+	}
+	return tags
+}
+
+type WeatherEntry struct {
+	Dt     int64  `json:"dt"`
+	Dttxt  string `json:"dt_txt"` // empty for weather/
+	Clouds struct {
+		All int64 `json:"all"`
+	} `json:"clouds"`
+	Main struct {
+		GrndLevel float64 `json:"grnd_level"` // empty for weather/
+		Humidity  int64   `json:"humidity"`
+		SeaLevel  float64 `json:"sea_level"` // empty for weather/
+		Pressure  float64 `json:"pressure"`
+		Temp      float64 `json:"temp"`
+		TempMax   float64 `json:"temp_max"`
```

Comment:
why not add TempMax and TempMin as fields well?

Reply:
I will test again, I don't remember why.

Reply:
OK found it, `temp_min` and `temp_max` are optional in the current weather API response. See the [interpretation](https://openweathermap.org/forecast16). These parameters may be applicable to large cities only.

>Please, do not confuse min/max parameters in current weather API and forecast API. In current weather API temp_min and temp_max are optional parameters mean min / max temperature in the city at the current moment to see deviation from current temp just for your reference. For large cities and megalopolises geographically expanded it might be applicable. In most cases both temp_min and temp_max parameters have the same volume as 'temp'. Please, use temp_min and temp_max parameters in current weather API optionally.

On the contrary, temp.min (not temp_min) and temp.max (not temp_max) given by the _forecast_ API are respectively the daily min temperature, and daily max temperature. Adding this info seems straightforward.

Reply:
We can make these struct fields pointers and then test if they are nil before adding.  This will allow them to be added if they exist.

Reply:
I've run more tests today. `temp_min` and `temp_max` are nearly equal to `temp` all the time, as suggested in OWM documentation.

I could not find in the API responses the temp {} structure mentioned in the docs.